scormxblock/scormxblock.py

```diff
@@ -1,21 +1,91 @@
-import json
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import mimetypes
 import re
-import os
 import pkg_resources
 import zipfile
-import shutil
 import xml.etree.ElementTree as ET
+from urllib.parse import urljoin, urlparse, unquote
+import boto3
+
+from os import path, walk
 
 from django.conf import settings
 from django.template import Context, Template
 from webob import Response
 
+from celery.task import task
+from fs.tempfs import TempFS
+from djpyfs import djpyfs
+
 from xblock.core import XBlock
 from xblock.fields import Scope, String, Float, Boolean, Dict
 from xblock.fragment import Fragment
 
+
 # Make '_' a no-op so we can scrape strings
 _ = lambda text: text
+FILES_THRESHOLD_FOR_ASYNC = getattr(settings, 'SCORMXBLOCK_ASYNC_THRESHOLD', 150)
+DEFAULT_CONTENT_TYPE = 'application/octet-stream'
+
+
+@task(name='scormxblock.scormxblock.s3_upload', routing_key=settings.HIGH_PRIORITY_QUEUE)
+def s3_upload(all_content, temp_directory, dest_dir):
+    """
+    Actual handling of the s3 uploads.
+    """
+    session = boto3.Session(
+        aws_access_key_id=settings.DJFS.get('aws_access_key_id'),
+        aws_secret_access_key=settings.DJFS.get('aws_secret_access_key'),
+        region_name=settings.DJFS.get('region_name'),
+    )
+    s3_client = session.resource('s3')
+    bucket = s3_client.Bucket(settings.DJFS.get('bucket'))
+
+    for filepath in all_content:
+        sourcepath = path.normpath(path.join(temp_directory.root_path, filepath))
+        destpath = path.normpath(path.join(dest_dir, filepath))
+        content_type = mimetypes.guess_type(sourcepath)[0]  # Returns a tuple
```

Comment:
```suggestion
        content_type = mimetypes.guess_type(sourcepath)[0]  or DEFAULT_CONTENT_TYPE  # Returns a tuple
```

instead of doing the if validation